api/scpca_portal/models/project.py

```diff
@@ -925,6 +925,12 @@ def update_spatial_data(future):
 
         non_downloadable_sample_ids = self.get_non_downloadable_sample_ids()
         max_workers = kwargs["max_workers"]
+
+        samples = (
+            Sample.objects.filter(project__scpca_id=self.scpca_id)
+            if sample_id is None
+            else Sample.objects.filter(scpca_id=sample_id).first()
```

Comment:
```suggestion
            else Sample.objects.filter(scpca_id=sample_id)
```

I think you want this to remain a list regardless. I believe adding `.first()` would give you back the one sample and not a list of samples

Reply:
Good call. If a `sample_id` is provided and I'll want to grab only that sample then it should still be housed within a list in order not to cause problems when calling `len()` and iterating over `for sample in samples` later on. Thanks for pointing this out.

To document, this ternary check is only for the short term, as we'll be removing the ability to pass an individual `sample_id` into `load_data` at some point soon in the future.